### PR TITLE
Store central state in a struct

### PR DESCRIPTION
**Pull request by philip-peterson**

### CODE REVIEW

File: StarryExpanse/Structs/RivenGameInstanceVars.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      InventoryBookPageTurnAmount = 0;
+      IsFrozenForLoading = false;
+      IsFrozenForLoadingFirstTime = false;
+    };
+
+  UPROPERTY(BlueprintReadOnly)
+    int32 CursorMode; // 0 = fixed to center, 1 = unlocked
+
+  UPROPERTY(BlueprintReadOnly)
+    int32 InventoryBookOpen; // 0 = none, 1 = atrus's journal, 2 = catherine's journal, 3 = trap book
```
Comment (cmumford) [inline StarryExpanse/Structs/RivenGameInstanceVars.h:30]: I'm assuming there's no easy way to make these enumerations?

File: StarryExpanse/Structs/RivenGameInstanceVars.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  UPROPERTY(BlueprintReadOnly)
+    int32 InventoryBookPage; // 0 = front cover, 1...N = different pages
+
+  UPROPERTY(BlueprintReadOnly)
+    float InventoryBookPageTurnAmount; // 0 = front cover, 1...N = different pages
```
Comment (cmumford) [inline StarryExpanse/Structs/RivenGameInstanceVars.h:36]: Now does this differ from InventoryBookPage aside from type?

Comment (philip-peterson) [reply StarryExpanse/Structs/RivenGameInstanceVars.h:36]: Yeah, sorry that comment was supposed to read differently. This is a fraction from 0..1 that indicates partially turned pages.

File: StarryExpanse/Structs/RivenGameInstanceVars.h (hunk 1 above)
Comment (philip-peterson) [reply StarryExpanse/Structs/RivenGameInstanceVars.h:30]: There is, but it's just more work and not really important to me right now.

**Pull request closed without merging**